configs: Check before use replacement policy options

abmerop opened the pull request:
Rather than adding the options to *every* config that might be using GPU_VIPER.py, just change the Ruby config to check if the option is available before trying to use it. Otherwise, reverts to what was the default on stable.

Change-Id: Ia6f1d0827d489ee2a35c598b644461cbff59e247

abmerop commented:
Should fix daily test failure.  https://github.com/gem5/gem5/actions/runs/9568053451/job/26400126071

mattsinc commented:
@abmerop I assume we'd need the same for the other replacement policy options like tcc-rp too?

@yuxiaojia : FYI

mattsinc commented on configs/ruby/GPU_VIPER.py:
```diff
@@ -303,7 +305,8 @@ def create(self, options):
             options.num_tccs, 2
         )
-        self.replacement_policy = RP_choose(options.tcp_rp)
+        if hasattr(options, "tcp_rp"):
```
I think this needs to be tcc_rp?

mattsinc commented on configs/ruby/GPU_VIPER.py:
```diff
@@ -241,7 +242,8 @@ class SQCCache(RubyCache):
         self.size = MemorySize(options.sqc_size)
         self.assoc = options.sqc_assoc
-        self.replacement_policy = RP_choose(options.tcp_rp)
+        if hasattr(options, "tcp_rp"):
```
I think this needs to be sqc_rp?

abmerop replied:
Yea, I think this was incorrect before

mattsinc replied:
Just to make sure I'm not going crazy: did you push a change to update this?  Or just add a comment?  If the former, I do not see it ...

abmerop replied:
Still working on it, should be up now

mattsinc reviewed (approved):
Thanks, @abmerop !  We'll have to be more careful about compiling many different things locally moving forward since I mistakenly though CI was testing GPU builds.

abmerop committed:
configs: Check before use replacement policy options

Rather than adding the options to *every* config that might be using
GPU_VIPER.py, just change the Ruby config to check if the option is
available before trying to use it. Otherwise, reverts to what was the
default on stable.

Change-Id: Ia6f1d0827d489ee2a35c598b644461cbff59e247
